client/layout/tracks-opt-out/index.jsx

```diff
@@ -0,0 +1,53 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import debug from 'debug';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { setTracksOptOut } from 'state/analytics/actions';
+
+/**
+ * Module variables
+ */
+const tracksDebug = debug( 'calypso:analytics:tracks' );
+
+class TracksOptOut extends React.Component {
+	static propTypes = {
+		userSettings: PropTypes.object.isRequired,
```

Comment:
Do we really need all userSettings, or can we pass through the opt out bool?

Reply:
We can pass through the opt-out bool, the "hard" part is making sure that we get a re-render when it's updated. Either we start `observe`-ing `userSettings` in the parent component ( `layout/index` ) or we observe inside `TracksOptOut` which seems like a more "contained" approche if we want to remove that `userSettings` in the future.